modules/e2ee/ManagedKeyHandler.js

```diff
@@ -0,0 +1,183 @@
+/* global __filename */
+
+import { getLogger } from 'jitsi-meet-logger';
+import debounce from 'lodash.debounce';
+
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+
+import { KeyHandler } from './KeyHandler';
+import { OlmAdapter } from './OlmAdapter';
+import { importKey, ratchet } from './crypto-utils';
+
+const logger = getLogger(__filename);
+
+// Period which we'll wait before updating / rotating our keys when a participant
+// joins or leaves.
+const DEBOUNCE_PERIOD = 5000;
+
+/**
+ * This module integrates {@link E2EEContext} with {@link OlmAdapter} in order to distribute the keys for encryption.
+ */
+export class ManagedKeyHandler extends KeyHandler {
+    /**
+     * Build a new AutomaticKeyHandler instance, which will be used in a given conference.
+     */
+    constructor(conference) {
+        super(conference);
+
+        this._key = undefined;
+        this._conferenceJoined = false;
+
+        this._olmAdapter = new OlmAdapter(conference);
+
+        this._rotateKey = debounce(this._rotateKeyImpl, DEBOUNCE_PERIOD);
+        this._ratchetKey = debounce(this._ratchetKeyImpl, DEBOUNCE_PERIOD);
+
+        // Olm signalling events.
+        this._olmAdapter && this._olmAdapter.on(
```

Comment:
no need to check for olmAdapter, it will exist for sure, you just created it a couple of lines above